src/plugins/__tests__/html.spec.ts

```diff
@@ -0,0 +1,158 @@
+import mergeHtml from '../mergeHtml';
+
+describe('merge HTML plugin', () => {
+  it('should handle empty element', () => {
+    const parsed = parse(`<span></span>`);
+
+    expect(
+      unified()
+        .use(mergeHtml)
+        .runSync(parsed),
+    ).toStrictEqual(
+      expect.objectContaining({
+        type: 'root',
+        children: [
+          {
+            type: 'paragraph',
+            position: expect.any(Object),
+            children: [
+              {
+                type: 'inlineHtml',
+                attributes: {},
+                tagName: 'span',
+                position: expect.any(Object),
+                children: [],
+              },
+            ],
+          },
+        ],
+      }),
+    );
+  });
+
+  it('should parse attributes', () => {
+    // based on https://github.com/nghiattran/html-attribute-parser/blob/master/test.js#L6
+    const parsed = parse(`<span attr class="animated infinite" ng-class="main.classAnimation"></span>`);
+
+    expect(
+      unified()
+        .use(mergeHtml)
+        .runSync(parsed),
+    ).toStrictEqual(
+      expect.objectContaining({
+        type: 'root',
+        children: [
+          {
+            type: 'paragraph',
+            position: expect.any(Object),
+            children: [
+              {
+                type: 'inlineHtml',
+                attributes: {
+                  attr: true,
+                  class: 'animated infinite',
+                  'ng-class': 'main.classAnimation',
+                },
+                tagName: 'span',
+                position: expect.any(Object),
+                children: [],
+              },
+            ],
+          },
+        ],
+      }),
+    );
+  });
+
+  it('should prepare children for inline html tag', () => {
+    const parsed = parse(`HTML link <a href="https://google.com">**hello** _world_ ~~navigate~~ to</a>.`);
+
+    expect(
+      unified()
+        .use(mergeHtml)
+        .runSync(parsed),
+    ).toStrictEqual(
+      expect.objectContaining({
+        type: 'root',
+        children: [
+          {
+            type: 'paragraph',
+            position: expect.any(Object),
+            children: [
+              {
+                type: 'text',
+                value: 'HTML link ',
+                position: expect.any(Object),
+              },
+              {
+                type: 'inlineHtml',
+                attributes: {
+                  href: 'https://google.com',
+                },
+                tagName: 'a',
+                position: expect.any(Object),
+                children: [
+                  {
+                    type: 'strong',
+                    position: expect.any(Object),
+                    children: [
+                      {
+                        type: 'text',
+                        value: 'hello',
+                        position: expect.any(Object),
+                      },
+                    ],
+                  },
```

Comment:
Are you sure we want to parse MD inside HTML? 

The [SMD docs](https://meta.stoplight.io/docs/studio-demo/docs/markdown/basic-syntax.md#inline-html) say it's not supported in HTML blocks (which the docs call "inline HTML"). I'd say it shouldn't be supported here either. Consistency matters with both our block-HTML parser and [Github's parser](https://github.com/stoplightio/studio-demo/blob/master/docs/markdown/basic-syntax.md#inline-html).

Reply:
> Are you sure we want to parse MD inside HTML?

I don't know. It'd be by far easier not to, but either way, we don't need to be strict here - the code will degrade gracefully anyway, so the core SMD principle is met, but we can punt on this and tell users to use HTML tags.

Reply:
Also, see this https://github.github.com/gfm/#example-138

Reply:
Having said that, I'm pretty sure we do need to support markdown inside _inline_ html elements.
markdown under `dl` (block element) won't be treated as plain text (same as we do right now), but
`<a href="foo.com">*foo*</a>`
<a href="foo.com">*foo*</a>
will not

Reply:
Alright, thanks for the explanation.